src/base/tristatebool.cpp

```diff
@@ -28,33 +28,67 @@
 
 #include "tristatebool.h"
 
-TriStateBool::TriStateBool()
-    : m_value(Undefined)
-{
-}
+#include <QDebug>
+#include <QJsonValue>
+
+const TriStateBool TriStateBool::Undefined(-1);
+const TriStateBool TriStateBool::False(0);
+const TriStateBool TriStateBool::True(1);
 
 TriStateBool::TriStateBool(bool b)
+    : m_value(b)
 {
-    m_value = (b ? True : False);
 }
 
-TriStateBool::TriStateBool(TriStateBool::ValueType value)
-    : m_value(Undefined)
+TriStateBool TriStateBool::fromInt(int i)
 {
-    switch (value) {
-    case Undefined:
-    case True:
-    case False:
-        m_value = value;
+    switch (i) {
+    case -1: return TriStateBool::Undefined; break;
```

Comment:
We have constants now but in another way. The old approach had problems with comparison operators.
What do you think, will it be better to use `boost::optional<bool>` instead?